[issue template] Emphasise that console output isn't enough (#26569)
microsoft/vcpkg: .github/ISSUE_TEMPLATE/report-package-build-failure.md

name: Report package build failure
about: Let us know about build failures in ports.
title: "[<port name>] build failure"
labels: ''
assignees: ''

---

**Host Environment**

- OS: [e.g. Windows/Linux etc...]
- Compiler: revision

**To Reproduce**

Steps to reproduce the behavior:

`./vcpkg install xxxx`

> ‼ **Note that you also have to provide the contents of *all* log files mentioned in the console output.** ‼

Reviewer: I'm not convinced a "also" note should come before the note it's meant to supplement.
I'd probably just add the emphasis added in the latest commit.

Author: I put this note intentionally above the `**Failure logs**` section because my observation is that people often replace the comments underneath `**Failure logs**`. If you browse through issues that have `requires:more-information`, you'll see what I mean.
Although I agree that this comment is somewhat redundant with the other comment below, I'd prefer to keep both because my hope is that people won't delete things that are above `**Failure logs**`.

The comment bellow describes more precisely what the user should do and how to do it whereas this comment is a reminder to do so and tries to explain why the log files are important.

That being said, I don't think it is sufficient to put more emphasis on the comment below because it is likely to get deleted without reading it anyways. In contrast, comments above the `**Failure logs**` section seem to get deleted less likely. If someone removes every comment below `**Failure logs**` and replaces it with their console output, the lines above will still be visible after the issue was created. My hope is that a user will then read them and add the contents of the log files.

Reviewer 2: I agree with @m-kuhn concern about including supplementary notes before the thing it supplements. I also agree with @autoantwort's suggestion to use the prefilled autogenerated issue body. Additionally, I would like to avoid redundancy whenever possible, and I believe this note and the one below emphasize the same thing.

‼ **The console output by itself is not enough to investigate a build failure.** ‼

**Failure logs**

- Cut and paste the appropriate build messages from the console output.

> - Attach ***every failure log files*** mentioned in the console output.

Reviewer: ```suggestion
- Attach ***every failure log file*** mentioned in the console output.
```

Typo from me already

> - Please attach any additional failure logs mentioned in the console output.

Reviewer: ```suggestion
- Attach any additional **failure log files** mentioned in the console output.
```

Let's be a bit more explicit here

Author: Thanks for your suggestion!

I replaced `additional` with `every` because I think that the term `additional` implies that the log files *just* provide additional information and that it is therefore not mandatory to provide them.

name: Report package build failure
about: Let us know about build failures in ports.
title: "[<port name>] build failure"

> labels: ''

Reviewer 3: ```suggestion
labels: category:port-bug
```

assignees: ''

---

> <!-- ⚠ Please use the generated issue template instead to report a port build failure. ⚠ -->
> <!-- ⁉ You can find its location at the end of the console output. ⁉ -->

Reviewer 3: ```suggestion
<!-- Please use the generated issue template instead to report a port build failure. -->
<!-- You can find its location at the end of the console output. -->
```
Can we remove these graphic characters?

Author: Why? What's wrong with using emojis?

Reviewer 3: I'm worried that these emoji won't display properly on some users' pages.

Reviewer 4: Some people just don't read texts... Emojis here are absolutely necessary.

Reviewer 3: Okay.
